internal/northbound/upload.go

```diff
@@ -212,16 +241,83 @@ func shouldValidateYAMLSchema(fileBytes []byte) bool {
 	return true
 }
 
+func (u *uploadSession) extractTarball(fileBytes []byte) (fileSet, error) {
+	contents := make(fileSet, 0)
+
+	// Convert fileBytes into an io.Reader
+	reader := bytes.NewReader(fileBytes)
+
+	// Create a gzip reader
+	gzipReader, err := gzip.NewReader(reader)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create gzip reader while extracting: %w", err)
+	}
+	defer gzipReader.Close()
+
+	// Create a tar reader
+	tarReader := tar.NewReader(gzipReader)
+
+	// Iterate through the files in the tar archive
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break // End of archive
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Failed to read tar header while extracting: %w", err)
+		}
+
+		// Check if the current file is the one we want to extract
+		if header.Typeflag == tar.TypeReg {
+			// Read the file content into a buffer
+			var buf bytes.Buffer
+			if _, err := io.CopyN(&buf, tarReader, MaxExtractedFileSize); err != nil && err != io.EOF {
+				return nil, fmt.Errorf("Failed to copy file content while extracting: %w", err)
```

Comment:
Should we clean contents map?  Likely chance that it may have some entries before we hit this error.

Reply:
go should garbage collect it automatically